CONTRIBUTING.md

```diff
@@ -0,0 +1,276 @@
+[developer issue here][developer-request].
+
+### Local Development
+
+#### Setup
+Follow the instructions laid out [here][lookit-setup-instructions].
+
+#### <a name="organization"></a> What's the high-level overview?
+Conceptually speaking, "Lookit-API" is a misnomer. Lookit is *not actually one service*, it is three (3) processes that
+act as (5) conceptually distinct systems/applications:
+1. The Django web app/http API process, providing:
+    - A web UI for researchers to design and manage experiments
+    - A web UI for participants to discover experiments
+    - An HTTP/JSON API consumed by built experiments
+2. An offline Celery process (connected by message queue) that builds and deploys experiments (web archives)
+3. An offline Celery process (connected by message queue) that sends emails and does some rudimentary cleanup
+
+The processes listed above also has three major *in-network* service dependencies:
+1. **Postgres** for relational data storage, used by all processes
+2. **RabbitMQ** as a message queue to communicate between process #1 and the two offline task runner processes
+3. **Docker** to provide a container environment for building experiments (used by process #2)
+
+And as if that wasn't enough, three major *out of network* service dependencies:
+1. **Amazon S3** for storing video
+2. **Google Cloud Storage** for storing experiment web archives and other static files
+3. [**Pipe**][pipe] to relay video streamed from experiments to Amazon S3
+
+For the kicker, if you want to debug the video renaming webhook, you'll need to install and run `ngrok`, and [add
+a new webhook target][pipe-add-webhook] in the Pipe web UI with the freshly generated URL. _Yikes_. Thankfully, you
+won't need to do this that often. Still - depending on how much you want to set up and debug locally, you could 
+potentially be running anywhere from 2-7 processes at once to support Lookit development.
+
+##### <a name="debugging"></a> OK this all sounds great - where do I point my debugger?
+
+If you want to debug one of the processes, you can always use PDB or the new `breakpoint` builtin. Alternatively, if
+you're using an IDE like PyCharm, you can set up handy debug profiles with neat conveniences like isolated environment
+variables and push-button functionality.
+
+If you just want to hack on the API or UIs, you only need to boot up Postgres and the web server (process #1):
+```bash
+brew services start postgresql  # Or `service postgresql start` if you're on linux
+./manage.py runserver           # Alternatively, `python manage.py runserver`
+```
+
+If you you want to run with local HTTPS (allowing webhook processing of videos), you'll need `runserver_plus` installed
+(should be in your virtual environment already if you ran `pipenv install --dev`) and a directory with cert files:
+```bash
+./manage.py runserver_plus --nopin --cert-file ${PATH_TO_YOUR_CERTFILE}
+```
```

Comment:
I think local HTTPS may be needed for smoothly running ember-lookit-frameplayer locally and having it talk to the local lookit-api instance at all at this point, not just the videos, because now browsers may balk at pulling e.g. study definitions & other data from the API over HTTP. But I haven't tested this in a while so if you're sure it's just the webhook for video processing that's affected ignore me.

Reply:
It's not configured by default, but I've added some language to indicate that the two need to match up.